### PR TITLE
Replace Sami by Doctum and re-build Docs

### PR DESCRIPTION
**Pull request by williamdes**

### CODE REVIEW

File: scripts/docs-template.html
Comment (williamdes) [inline scripts/docs-template.html:1]: could be improved if the PR is okay

**Comment by williamdes** [comment]
Test: `doctum.phar update scripts/doctum.php --force && SKIP_PUSH=1 ./scripts/update-built-docs.sh`

**Comment by williamdes** [comment]
PHP 8 fix in progress Upstream

**Comment by williamdes** [comment]
The fix works, will do a release for it soon.
Working on the permission issue with the docker image

**Comment by OndraM** [comment]
From the build output it seems the lint-docs runs even on the previous releases. Is this intentional? I'd expect it to report only errors from for example the latest commit.

Also - it found an issue `The method "createBySessionID" has "5" @param tags but only "4" where expected. on "Facebook\WebDriver\Remote\RemoteWebDriver::createBySessionID"` error. But this is intentional to keep BC, there is `func_num_args` condition in body od the method. Is is possible to mark this error ignored?

**Comment by williamdes** [comment]
> From the build output it seems the lint-docs runs even on the previous releases. Is this intentional? I'd expect it to report only errors from for example the latest commit.

Yes, this was also the Sami behavior

> Also - it found an issue `The method "createBySessionID" has "5" @param tags but only "4" where expected. on "Facebook\WebDriver\Remote\RemoteWebDriver::createBySessionID"` error. But this is intentional to keep BC, there is `func_num_args` condition in body od the method. Is is possible to mark this error ignored?

Yeah, `--ignore-parse-errors` removes the exit code and that allows errors to be ignored

**Comment by williamdes** [comment]
Variables to define in an actions environnement named: `API documentation`
- `SSH_KEY_DEPLOY` (the SSH private key you generated, put the public key as a deploy key with write access)
- `GIT_COMMIT_AUTHOR` (`John Doe <foo@bar.tld>`)

**Comment by williamdes** [comment]
Status: publishing works, lint works.
Waiting for upstream release to switch actions from `dev` to `v5` channel

File: scripts/update-built-docs.sh
Comment (OndraM) [inline scripts/update-built-docs.sh:29]:
```suggestion
    git diff-index --quiet HEAD || git commit -m "Api documentations update ($(date --rfc-3339=seconds --utc))" -m "#apidocs" && if [ -z "${SKIP_PUSH}" ]; then git push; fi
```

To be sure its not a locale-specific date.
```sh
$ date --utc
Čt 22. dubna 2021, 11:32:49 UTC
$ date --rfc-3339=seconds --utc
2021-04-22 11:32:53+00:00
```

**Review by OndraM — CHANGES_REQUESTED** [review]
Nice job! 

So this means I should create a new github user for this purpose, and add the build secrets as you wrote above?

BTW -aren't we somehow able to get rid of the lint errors from the pull request diff page ("Unchanged files with check annotations")? It would be a bit confusing for any new pull request 🤔

obrazek

File: .github/workflows/publish-docs.yml
Comment (williamdes) [reply .github/workflows/publish-docs.yml:4]: When you go to the workflow you will have a button to trigger manually the job

File: .github/workflows/publish-docs.yml
Comment (williamdes) [reply .github/workflows/publish-docs.yml:7]: this is a little overkill and resource wasting IMO
Also I think it is better to be ran in it's isolated context

File: scripts/update-built-docs.sh
Comment (williamdes) [reply scripts/update-built-docs.sh:29]: Nice, I will apply that

**Comment by williamdes** [comment]
> Nice job!
> 
> So this means I should create a new github user for this purpose, and add the build secrets as you wrote above?
> 
> BTW -aren't we somehow able to get rid of the lint errors from the pull request diff page ("Unchanged files with check annotations")? It would be a bit confusing for any new pull request 
> 

No need to create a new user, you just need to decide the name of the committer and generate a SSH keypair for the build step to push the changes to the branch

> BTW -aren't we somehow able to get rid of the lint errors from the pull request diff page ("Unchanged files with check annotations")? It would be a bit confusing for any new pull request 

Not for now, but you can open an issue on Doctum and I will make a fix for that. "Frozen" tags should not report errors.
Still did not decide if I should add a CLI flag to remove the errors or just remove them for "Frozen" tags 🤔

**Comment by williamdes** [comment]
Sélection_2021042213:58:33001
Sélection_2021042213:58:15001
Sélection_2021042213:57:55001

**Comment by OndraM** [comment]
> "Frozen" tags should not report errors. Still did not decide if I should add a CLI flag to remove the errors or just remove them for "Frozen" tags thinking

This would be much better, but still the error in `createBySessionID` would persist...

**Comment by williamdes** [comment]
I am lacking some time to focus on this PR but it is open in my browser as a tab so I will finish this some day soon

**Comment by williamdes** [comment]
> > I am lacking some time to focus on this PR but it is open in my browser as a tab so I will finish this some day soon
> 
> No problem, I marked it as draft - I guess you should have rights to "unmark" it once ready.

I will try to allocate a day this week for this task

**Comment by williamdes** [comment]
> > It looks nice! I'm only worried about the errors which will it probably add to every following pull request :-( . Isn't there any way how to suppress this?
> 
> Ha, I had forgot about it I did open an issue for that: [code-lts/doctum#33](https://github.com/code-lts/doctum/issues/33) Should be a quick fix, will try to do that soon

The version [5.5.0](https://github.com/code-lts/doctum/releases/tag/v5.5.0) is out 🎉 
Errors will not be in the output anymore

**Comment by williamdes** [comment]
Hi @OndraM 
Do you think this one could be merged before the next tag ?
I am okay to wait more if you do not have time :)

**Comment by OndraM** [comment]
@williamdes Hi, I made same changes to the actions and [pushed them into my fork](https://github.com/OndraM/php-webdriver/commit/7906bbd40bbbadc11bcabc7edad49e1c6e4b9a2b).

Could you have a look and cherry pick them to your fork, in order to update the PR? (I don't have permission to update your fork.)

For example I don't want it to commit automatically on behalf of the specific user but rather by some "Automated" user, see the commit.

**Comment by williamdes** [comment]
> @williamdes Hi, I made same changes to the actions and [pushed them into my fork](https://github.com/OndraM/php-webdriver/commit/7906bbd40bbbadc11bcabc7edad49e1c6e4b9a2b).
> 
> Could you have a look and cherry pick them to your fork, in order to update the PR? (I don't have permission to update your fork.)
> 
> For example I don't want it to commit automatically on behalf of the specific user but rather by some "Automated" user, see the commit.

Sure, I picked the commit

Sorry for the noise my force pushs did make in notifications

I added de0b3f0714a0a48f48b3c02c34e075dc27141105 to bump actions/checkout in this PR too

**Review by OndraM — APPROVED** [review]

**Comment by OndraM** [comment]
Merged and deployed, thanks @williamdes! :tada: 

http://php-webdriver.github.io/php-webdriver/main/

**Comment by williamdes** [comment]
> Merged and deployed, thanks @williamdes! tada
> 
> http://php-webdriver.github.io/php-webdriver/main/

Thank you too ! 💯 
I did open #977 in case you could push a commit for it, I deleted my fork before seeing this mistake

**Pull request merged**
